include/dem/insertion.h

```diff
@@ -90,18 +95,68 @@ class Insertion
    * @param inserted_this_step Number of particles that are inserted
    * at each insertion step. This value can change in the last insertion step to
    * reach the desired number of particles
+   * @param current_inserting_particle_type Type of inserting particles
    */
   std::vector<std::vector<double>>
```

Comment:
It is a vector of vector, since it stores all properties of all particles. This function is called once during each insertion step. Should I define it as an input argument?

Reply:
I would at least use move semantic or something to prevent copying of such a large vector. You have two choices, either use move semantic or pass an argument to that function which is the reference of the vector. I suggest you use move semantic however.